### PR TITLE
OCPBUGS-42737: add nil check to etcdRecoveryActiveCondition when coll…

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
**What this PR does / why we need it**: currently HO crashes due to nil pointer  on upgrades because we try to retrieve a status condition when its not available.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes # [OCPBUGS-42737](https://issues.redhat.com/browse/OCPBUGS-42737)

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by enxebre** [comment]
Please include a validation for this in the existing ValidateMetrics https://github.com/openshift/hypershift/blob/cde9ea443a8ec18b3a436de570c96cecab06b8c2/test/e2e/util/util.go#L1344

File: hypershift-operator/controllers/hostedcluster/metrics/metrics_test.go [hunk 1]
```diff
@@ -765,6 +765,102 @@ func TestReportDeletingDuration(t *testing.T) {
 	}
 }
 
+func TestReportEtcdManualInterventionRequired(t *testing.T) {
+	wrapExpectedValueAsMetric := func(expectedValue float64) *dto.MetricFamily {
+		return &dto.MetricFamily{
+			Name: pointer.String(EtcdManualInterventionRequiredMetricName),
```
Comment (jparrill) [inline hypershift-operator/controllers/hostedcluster/metrics/metrics_test.go:771]: pointer.String is deprecated already, I recommend you to move to `k8s.io/utils/ptr` lib with ptr.To()

**Comment by enxebre** [comment]
Thanks!
/label tide/merge-method-squash
/lgtm
/hold cancel

**Pull request merged**